### PR TITLE
contextless macros

### PR DESCRIPTION
**Pull request by cmcarthur**
WIP

### CODE REVIEW

File: test/unit/test_parser.py [hunk 1]
```diff
@@ -1138,6 +1138,22 @@ def test__simple_macro(self):
...
+{%- endmacro %}
+"""
+
+        with self.assertRaises(dbt.exceptions.CompilationException):
+            dbt.parser.parse_macro_file(
+                macro_file_path='macro_with_ref.sql',
+                macro_file_contents=macro_file_contents,
+                root_path=get_os_path('/usr/src/app'),
+                package_name='root')
+
```
Comment (cmcarthur) [inline test/unit/test_parser.py]: @drewbanin fyi

File: dbt/clients/jinja.py [hunk 2]
```diff
@@ -3,10 +3,44 @@
...
+                        self.onError(token)
+
+                elif token.test('name') and token.value in self.DisallowedVars:
+                    self.onError(token)
+
+                for token in held:
+                    yield token
+
+    return jinja2.sandbox.SandboxedEnvironment(
+        extensions=[MacroContextCatcherExtension])
```
Comment (cmcarthur) [inline dbt/clients/jinja.py]: so. cool.

**Comment by drewbanin** [comment]
@cmcarthur this should also check if `this`, `ref`, `target`, etc are passed in as args, right? There shouldn't be a compiler error if `this` is passed in as an arg!

**Comment by cmcarthur** [comment]
@drewbanin yep, that's right. anything passed in as an arg can be removed from the disallowed lists

**Comment by cmcarthur** [comment]
merged into #356

**Pull request closed without merging**